src/mlpack/methods/random_forest/bootstrap.hpp

```diff
@@ -18,30 +18,229 @@ namespace mlpack {
+      arma::vec&        invConcurrency,
+      const IndMatType& intervals)
+  {
+    // phi may have less rows than intervals has columns as
+    // phi is grown in multiple steps.
+    assert(phi.n_rows <= intervals.n_cols && phiSize < phi.n_rows);
+
+    arma::vec avg(intervals.n_cols);
+
+    for (arma::uword i(0); i < avg.size(); ++i)
+    {
+      // Temporarily assume another concurrency for the current i.
+      concurrency.subvec(intervals(0, i), intervals(1, i)) += 1.0;
+      invConcurrency.subvec(intervals(0, i), intervals(1, i)) =
+          1.0 / concurrency.subvec(intervals(0, i), intervals(1, i));
+
+      phi[phiSize] = i;
+      avg[i] = ComputeAverageUniqueness(
+          intervals.cols(phi.rows(0, phiSize)),
+          phiSize,
+          concurrency,
+          invConcurrency);
+
+      concurrency.subvec(intervals(0, i), intervals(1, i)) -= 1.0;
+      invConcurrency.subvec(intervals(0, i), intervals(1, i)) =
+          1.0 / concurrency.subvec(intervals(0, i), intervals(1, i));
+    }
+
+    return avg / arma::sum(avg);
+  }
+
+  /**
+   * Compute the samples of the next draw.
+   *
+   * @param[in] colCount Number of data points in the dataset and
+   *                     the number of samples to draw respectively.
+   *
+   * @return A list of indices referring to the observations that should
+   *         be sampled.
+   */
+  arma::uvec ComputeSamples(arma::uword colCount) const
+  {
+    // Randomly sample columns from the intervals matrix.
+    DiscreteDistribution d(intervals.n_cols);
+    arma::uvec           phi(colCount);
+    arma::vec            concurrency(intervals.max() + 1, arma::fill::zeros);
+    arma::vec            invConcurrency(intervals.max() + 1, arma::fill::ones);
+
+    for (arma::uword i(0); i < colCount; ++i)
+    {
+      d.Probabilities() = ComputeNextDrawProbabilities(
+          phi, i, concurrency, invConcurrency, intervals);
+      assert(d.Probabilities().size() == intervals.n_cols);
+
+      phi[i] = d.Random()[0];
+      concurrency.subvec(intervals(0, phi[i]), intervals(1, phi[i])) += 1.0;
+      invConcurrency.subvec(intervals(0, phi[i]), intervals(1, phi[i])) =
+          1.0 / concurrency.subvec(intervals(0, phi[i]), intervals(1, phi[i]));
+    }
+
+    return phi;
+  }
+
+  template<
+      bool UseWeights,
+      typename MatType,
+      typename LabelsType,
+      typename WeightsType>
+  void Bootstrap(
+      const MatType&     dataset,
+      const LabelsType&  labels,
+      const WeightsType& weights,
+      MatType&           bootstrapDataset,
+      LabelsType&        bootstrapLabels,
+      WeightsType&       bootstrapWeights)
+  {
+    if (labels.n_cols != intervals.n_cols)
+      throw std::invalid_argument("SequentialBootstrap::Bootstrap(): "
+          "labels n_cols and intervals n_cols differ!");
+
+    // observations are stored as columns and dimensions
+    // (number of features) as rows.
+    const arma::uvec phi(ComputeSamples(dataset.n_cols));
```

Comment:
```suggestion
    const arma::uvec phi = ComputeSamples(dataset.n_cols);
```
This is pedantic, and you can choose to ignore this comment if you prefer, but the use of `=` instead of parenthetical-initialization throughout will match the rest of the mlpack codebase better.